src/lib/controller/message_handler.ts

```diff
@@ -118,3 +153,59 @@ export class ControllerMessageHandler {
     }
   }
 }
+
+function processInclusionOptions(
+  clientsController: ClientsController,
+  client: Client,
+  message: IncomingMessageController
+): InclusionOptions | ReplaceNodeOptions {
+  // Schema 8+ inclusion handling
+  if ("options" in message) {
+    if (
+      message.options.strategy === InclusionStrategy.Default ||
+      message.options.strategy === InclusionStrategy.Security_S2
+    ) {
+      message.options.userCallbacks = {
+        grantSecurityClasses: (
+          requested: InclusionGrant
+        ): Promise<InclusionGrant | false> => {
+          clientsController.grantSecurityClassesPromise =
+            createDeferredPromise();
+          client.sendEvent({
+            source: "controller",
+            event: "grant security classes",
+            requested: requested as any,
+          });
+          return clientsController.grantSecurityClassesPromise;
```

Comment:
Would you be able to schedule to clean up the promises here. That way you are setting and cleaning it up at the same location.

```suggestion
          clientsController.grantSecurityClassesPromise.finally(() => {
              clientsController.grantSecurityClassesPromise;
          }); 
          return clientsController.grantSecurityClassesPromise;
```